lib/zclient.c

```diff
@@ -3341,3 +3386,83 @@ int32_t zapi_capabilities_decode(struct stream *s, struct zapi_cap *api)
 stream_failure:
 	return 0;
 }
+
+int zclient_send_srv6_function(struct zclient *zclient,
+			       const struct srv6_function *function)
+{
+	struct stream *s;
+
+	s = zclient->obuf;
+	stream_reset(s);
+	zapi_srv6_function_encode(ZEBRA_SRV6_FUNCTION_ADD, s, function);
+	zclient_send_message(zclient);
+	return 0;
+}
+
+int zapi_srv6_locator_encode(uint8_t cmd, struct stream *s,
+			     const struct srv6_locator *locator)
+{
+	uint8_t namelen;
+
+	zclient_create_header(s, cmd, VRF_DEFAULT);
+	namelen = strlen(locator->name);
+	stream_putc(s, namelen);
+	stream_put(s, locator->name, namelen);
+	stream_put(s, &locator->prefix, sizeof(locator->prefix));
+	stream_putc(s, locator->function_bits_length);
+	stream_putw_at(s, 0, stream_get_endp(s));
+	return 0;
+}
+
+int zapi_srv6_locator_decode(struct stream *s, struct srv6_locator *locator)
+{
+	uint8_t namelen;
+
+	STREAM_GETC(s, namelen);
+	stream_get(locator->name, s, namelen);
+	locator->name[namelen] = '\0';
+	stream_get(&locator->prefix, s, sizeof(locator->prefix));
+	locator->function_bits_length = stream_getc(s);
+stream_failure:
```

Comment:
unnecessary label. Could be remove

Reply:
Not true, please look at `stream.h`

Reply:
label `stream_failure` is used by macro STREAM_GETC.